src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Compilation.cs

```diff
@@ -264,7 +264,8 @@ public bool NeedsRuntimeLookup(ReadyToRunHelperId lookupKind, object targetOfLoo
         public ReadyToRunHelperId GetLdTokenHelperForType(TypeDesc type)
         {
             bool canConstructPerWholeProgramAnalysis = _devirtualizationManager == null ? true : _devirtualizationManager.CanConstructType(type);
-            return canConstructPerWholeProgramAnalysis & DependencyAnalysis.ConstructedEETypeNode.CreationAllowed(type)
+            bool creationAllowed = DependencyAnalysis.ConstructedEETypeNode.CreationAllowed(type);
+            return canConstructPerWholeProgramAnalysis & creationAllowed
```

Comment:
```suggestion
            return (canConstructPerWholeProgramAnalysis && creationAllowed)
```
These are bools. Logical AND is more appropriate. 

I am actually surprised that bitwise AND on bools is allowed in C#.

Reply:
Yeah, typo, this is a bitwise `and`. I wasn't sure what the intent was here, so I was keeping the semantics the same, If logical AND is appropriate, I will switch to that.